### ISSUE TITLE
Skipped 163140 files listed in gphotos.bad_ids

### ISSUE THREAD

**Issue by computerartclub** [gilesknap/gphotos-sync#313]
**Describe the bug**
Try to sync on a raspberry pi 4 running Ubuntu 20.10 using gphotos-sync 2.14.2. Gphotos-sync was installed using snap. 
The result is missing most of my older photos, most of the older albums are empty(everything from 2010-2019 I believe). Also, when I try to open the pictures from the albums folder on my windows machine I just get error messages saying the files are corrupt. In total, only 200 gigabytes out of 1000 gigabytes of data downloaded (though I am sure video compression plays a large part too). 75059 pictures were downloaded. 163140 were skipped.
 
Some of my attempts have included: 
--retry-download (about 2-3 times)
--max-retries 5 --progress (I think this is what generated the bad IDs file?)

I think I have also tried
--rescan

I checked a previous thread about bad_ids but I believe the conclusion was that Google had fixed the error and the bad_ids.yaml file was no longer necessary as a result. 

Any suggestions?

**Comment by gilesknap** [comment]
I'm afraid that new issues with the Google API emerged and I reinstated the bad IDs feature. 

With luck the issues you have had are transient in which case deleting the bad IDs file and trying again (default arguments) may help.

If not I'll get you to create a log and send it to me.

**Comment by computerartclub** [comment]
OK. Thank you! So after deleting the bad IDs file and running the script a few more times with similar parameters as before, it ended up downloading what looks to be all the photos and videos (now 567 GB worth, but the difference is likely in the video compression).

However, I am not sure exactly how the albums folder works. In each folder there are small files with similar file names to the original photos, but each file has a numerical prefix (perhaps indicating its position in the album). The files are roughly 100 bytes each and when I try to open them on  a windows machine it tells me that 'Windows Photo Viewer can't open this picture because the file appears to be damaged, corrupted or is too large'. 
Are these supposed to be shortcuts to the images? Or preview images? Placeholders? Do they work differently for you or with a different OS?  Or do they just give some sort of an indication of the original makeup of these folders without actually putting the files in the folders themselves?

**Comment by gilesknap** [comment]
The album files are supposed to be links to the original files. There are
some instructions for enabling links on windows somewhere on the repository
(probably the readme but I'm away from my PC so can't check).

**Comment by computerartclub** [comment]
Thanks for your reply, I see now that it says the files are symlinks and in another section of the readme it has a link for enabling symlinks in windows: https://community.perforce.com/s/article/3472
I tried following along but it didn't really change anything, further internet searches made me suspect that they are not very windows friendly. I just tried it again on Ubuntu however and it seems that they are more Linux friendly and they seem to work there.

Thank you for your time and a great script! :)
